docs/tutorials/reactjs-tutorial.md

```diff
@@ -0,0 +1,273 @@
+
+>**Note**: If you've previously installed `create-react-app` globally via `npm install -g create-react-app`, we recommend you uninstall the package using `npm uninstall -g create-react-app` to ensure that npx always uses the latest version.
+
+Let's quickly run our React application by navigating to the new folder and typing `npm start` to start the web server and open the application in a browser:
+
+```bash
+cd my-app
+npm start
+```
+
+>Note: in order to avoid the following [issue](https://github.com/eclipse-theia/theia/issues/6720#issuecomment-563488535) you have to add `jsconfig.json` in the top of the folder.
+
+``` json
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "target": "es2016",
+        "jsx": "preserve",
+        "allowJs": true,
+        "allowSyntheticDefaultImports": true,
+        "allowNonTsExtensions": true
+    }
+}
+```
+
+<!-- TBD mention yarn and link -->
+
+You should see the React logo and a link to "Learn React" on [http://localhost:3000](http://localhost:3000) in your browser. We'll leave the web server running while we look at the application with Emacs.
+
+To open your React application in Emacs, open another terminal or command prompt window, navigate to the `my-app` folder and type ``:
+
+>**Tip**: For optimal experience use either `Emacs 27.1` or [gccemacs](https://akrl.sdf.org/gccemacs.html).
+
+### Emacs configuration
+
+Here it is minimalistic sufficient configuration to run this tutorial with:
+
+``` emacs-lisp
+(package-initialize)
+(add-to-list 'package-archives '("melpa" . "http://melpa.org/packages/") t)
+(setq package-selected-packages '(lsp-mode yasnippet lsp-treemacs helm-lsp projectile hydra flycheck company avy which-key helm-xref dap-mode zenburn-theme json-mode))
+(when (cl-find-if-not #'package-installed-p package-selected-packages)
+  (package-refresh-contents)
+  (mapc #'package-install package-selected-packages))
+(load-theme 'zenburn t)
+(helm-mode)
+(require 'helm-xref)
+(define-key global-map [remap find-file] #'helm-find-files)
+(define-key global-map [remap execute-extended-command] #'helm-M-x)
+(define-key global-map [remap switch-to-buffer] #'helm-mini)
+(which-key-mode)
+(add-hook 'prog-mode-hook #'lsp)
+(setq gc-cons-threshold (* 100 1024 1024)
+      read-process-output-max (* 1024 1024)
+      company-idle-delay 0.0
+      company-minimum-prefix-length 1
+      create-lockfiles nil ;; lock files will kill `npm start'
+      lsp-headerline-breadcrumb-enable t)
+(with-eval-after-load 'lsp-mode
+  (require 'dap-chrome)
+  (add-hook 'lsp-mode-hook #'lsp-enable-which-key-integration)
+  (yas-global-mode))
+```
+
+```bash
+emacs my-app/src/index.js
+```
+
+### Installing typescript language server
+
+`Emacs` supports automatic intallation of the language servers so the first time you start emacs in javascript file and you run `M-x lsp` you will be asked to select the language server to install. Select `ts-ls`.
```

Comment:
```suggestion
`Emacs` supports automatic language server installation, so the first time you run `M-x lsp` in a JavaScript file opened in it, you will be prompted for a language server to install. Select `ts-ls`.
```

Also, is `Emacs` the right term here? It isn't actually Emacs that does the server installation, but `lsp-mode`. `eglot`, for example, [doesn't seem to have that](https://github.com/joaotavora/eglot/issues/69). I understand that newcomers might not know/care about the distinction, so if they are the target audience, this is probably better.